### PR TITLE
Run scheduled job to fetch data

### PR DESCRIPTION
**Pull request by palexcast**
The npm command `fetch-data` fetches new data for all municipalities.
If any municipality should be missing data, it will fetch the most recent data.

This results in a json file with history of all the data aggregated, and data[0] will contain as much data as possible.

The build command will read the most recent of these.

There is also a new job that you can trigger to run this job, which also deploys the site, as it won't trigger the eleventy workflow (it is also more efficient to run in 1 job).

Know this might not be the "store" solution you were looking for, but this can easily be moved to a db or something similar later 🍡 

Just a proposal, so comment away! Code might not be the cleanest 😅

Runs: https://github.com/palexcast/green-municipalities/actions

### CODE REVIEW

**Comment by boyum** [comment]
`.github/workflows/eleventy.yml` no longer exists, please resolve by deleting it ✨

File: .github/workflows/fetch-data.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+      - uses: bahmutov/npm-install@v1
+
+      - name: Build
+        run: npm run fetch-data
+
+      - name: Update Data
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Updated Data
```
Comment (boyum) [inline .github/workflows/fetch-data.yml:25]:
```suggestion
          commit_message: fetch daily carbon data
```

File: .github/workflows/fetch-data.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Build
+        run: npm run build
+
```
Comment (boyum) [inline .github/workflows/fetch-data.yml:29]:
```suggestion
```

**Comment by boyum** [comment]
Perhaps we should limit the daily fetcher to run on main, not on PRs? Or will it handle itself automatically if there are more branches?

**Comment by palexcast** [comment]
> Perhaps we should limit the daily fetcher to run on main, not on PRs? Or will it handle itself automatically if there are more branches?

It runs a scheduled job, not matter the branch. Jobs are not directly related to branches.
The first step is checkout, with a ref: main, så this means it will always checkout main.

**Comment by palexcast** [comment]
@boyum Ran it locally, it now contains 354 municiples 😍 Nice catch!

**Review by boyum — APPROVED** [review]
🥳

**Pull request merged**